### ISSUE TITLE
HTML headings and tree section

### ISSUE THREAD

**Issue by matses** [mfuchs23/trafo#1]
Hi,

I suspect a bug in the sections tree generated by Herold, when 2 html headings share the same level : 

eg with h2 :

``` html
<h1>title1</h1><p>para 1</p>
<h2>title 2.1</h2><p>para 2.1</p>
<h2>title 2.2</h2><p>para 2.2</p>
```

converted in : 

``` html
<section>
    <title> title1 </title>
    <para>para 1</para>
    <section>
      <title>title 2.1</title>
      <para>para 2.1</para>
      <section>
        <title>title 2.2</title>
        <para>para 2.2</para>
      </section>
    </section>
  </section>
```

In this case, i expected section 2.1 and section 2.2 at the same level in the tree sections :

``` html
<section>
    <title> title1 </title>
    <para>para 1</para>
    <section>
      <title>title 2.1</title>
      <para>para 2.1</para>
    </section>      
     <section>
        <title>title 2.2</title>
        <para>para 2.2</para>      
    </section>
  </section>
```

Do you agree?

**Comment by mfuchs23** [comment]
Hello, I agree. Which version of herold did you use?

**Comment by matses** [comment]
6.1.0-188. 
The jar is from herold-6.1.0-188.tar.gz available at http://www.dbdoclet.org/index.html

**Comment by mfuchs23** [comment]
Fixed with release 8.0.0

**Comment by matses** [comment]
Thanks! 
Is the jar available elsewhere or do i have to build it from sources by myself?

**Comment by mfuchs23** [comment]
You can download installation packages from http://www.dbdoclet.org

**Comment by mfuchs23** [comment]
You can download installation packages from http://www.dbdoclet.org

Am 23.11.2015 um 10:18 schrieb matses:

**Comment by matses** [comment]
OK many thanks.
I made some tests this morning... Works well for the above example

But doesn't work for this one : 

``` html
<p>para 1</p>
<h2>title 2.1</h2>
<p>para 2.1</p>
<h2>title 2.2</h2>
<p>para 2.2</p>
```

--> 

``` html
<para>para 1</para>
<section>
  <title>title 2.1</title>
  <para>para 2.1</para>
  <section>
    <title>title 2.2</title>
    <para>para 2.2</para>
  </section>
</section>
```

My example is not very HTML5 compliant, because of the lack of h1, but it is a fragment...

**Comment by mfuchs23** [comment]
OK, thank you for testing. I will fix it as soon as possible.

**Comment by mfuchs23** [comment]
Hello, this issue is fixed with release 8.0.1

Am 23.11.2015 um 11:57 schrieb matses:

> OK many thanks.
> I made some tests this morning... Works well for the above example
> 
> But doesn't work for this one :
> 
> <p>para 1</p>
> 
> <h2>title 2.1</h2>
> 
> <p>para 2.1</p>
> 
> <h2>title 2.2</h2>
> 
> <p>para 2.2</p>
> 
> -->
> 
> <para>para 1</para>
> <section>
>   <title>title 2.1</title>
>   <para>para 2.1</para>
>   <section>
>     <title>title 2.2</title>
>     <para>para 2.2</para>
>   </section>
> </section>
> 
> My example is not very HTML5 compliant, because of the lack of h1, but
> it is a fragment...
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mfuchs23/trafo/issues/1#issuecomment-158904305.

## 

Tel. 0160 7845384
michael.fuchs@dbdoclet.org
http://www.dbdoclet.org
